src/PlaywrightSharp/Mouse.cs

```diff
@@ -13,16 +12,26 @@ public Mouse(PageChannel channel)
             _channel = channel;
         }
 
-        public Task ClickAsync(decimal x, decimal y, int delay = 0, MouseButton button = MouseButton.Left, int clickCount = 1)
-            => _channel.MouseClickAsync(x, y, delay, button, clickCount);
+        public Task ClickAsync(float x, float y, MouseButton button, int? clickCount, float? delay)
+            => _channel.MouseClickAsync(x, y, delay ?? 0, EnsureDefaultValue(button), clickCount ?? 1);
 
-        public Task DblClickAsync(decimal x, decimal y, int delay = 0, MouseButton button = MouseButton.Left)
-            => _channel.MouseClickAsync(x, y, delay, button, 2);
+        public Task DblclickAsync(float x, float y, MouseButton button, float? delay)
+            => _channel.MouseClickAsync(x, y, delay ?? 0, EnsureDefaultValue(button), 2);
 
-        public Task DownAsync(MouseButton button = MouseButton.Left, int clickCount = 1) => _channel.MouseDownAsync(button, clickCount);
+        public Task DownAsync(MouseButton button, int? clickCount)
+            => _channel.MouseDownAsync(EnsureDefaultValue(button), clickCount ?? 1);
 
-        public Task MoveAsync(decimal x, decimal y, int? steps = 1) => _channel.MouseMoveAsync(x, y, steps);
+        public Task MoveAsync(float x, float y, int? steps)
+            => _channel.MouseMoveAsync(x, y, steps ?? 1);
 
-        public Task UpAsync(MouseButton button = MouseButton.Left, int clickCount = 1) => _channel.MouseUpAsync(button, clickCount);
+        public Task UpAsync(MouseButton button, int? clickCount)
+            => _channel.MouseUpAsync(EnsureDefaultValue(button), clickCount ?? 1);
+
+        private static MouseButton EnsureDefaultValue(MouseButton button) =>
+            button switch
+            {
+                MouseButton.Undefined => MouseButton.Left,
```

Comment:
Maybe fail with an `InvalidArgumentException`?

Reply:
Why would we fail? If it's `Undefined` (which is `default`) we want to default back to `left`. This is technically a problem caused by the fact that we add the `Undefined` in there, but we don't specify the actual default in the API specs. This is something we can potentially think of doing in the future. But either way, I wouldn't throw, *just yet*.